Adds VideoDecoder (#8984)
mdn/content: files/en-us/web/api/videodecoder/videodecoder/index.md

---
title: VideoDecoder.VideoDecoder()
slug: Web/API/VideoDecoder/VideoDecoder
tags:
  - API
  - Constructor
  - Reference
  - VideoDecoder
browser-compat: api.VideoDecoder.VideoDecoder
---
{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}

> The **`VideoDecoder()`** constructor creates a new {{domxref("VideoDecoder")}} object with the provided `init.output` callback assigned as the output callback, the provided `init.error` callback as the error callback, and the {{domxref("VideoDecoder.state")}} set to `"unconfigured"`.

Reviewer: I'm not crazy about describing the init members in the summary when they are described in a dedicated section. What's the argument for it?

Author: I feel these intros should be a summary of what's going on I guess, but I'm not altogether bothered either way. I don't think we describe anywhere else that the video `state` is set to unconfigured which is important as you have to both create the decoder and configure it, where with some things you create an object and it's ready to go.

## Syntax

```js
new VideoDecoder();
```

### Parameters

- `init`

>   - : A dictionary object containing two required callbacks.

Reviewer: ```suggestion
  - : A dictionary object containing two callbacks.
```

    - `output`

>       - : A callback which is passed a single argument of an {{domxref("VideoFrame")}} object.

Reviewer: ```suggestion
      - : A callback which takes a {{domxref("VideoFrame")}} object as its only argument.
```

    - `error`

>       - : A callback which is passed a single argument of the error that has been thrown.

Reviewer: ```suggestion
      - : A callback which takes an {{jsxref("Error")}} object as its only argument.
```

## Examples

> In the following example an `VideoDecoder` is created with the two required callback functions, one to deal with the decoded frame and the other to handle errors.

Reviewer: ```suggestion
In the following example a `VideoDecoder` is created with the two required callback functions, one to deal with the decoded frame and the other to handle errors.
```
